.github/workflows/ci.yaml

```diff
@@ -40,4 +40,26 @@ jobs:
 
     - name: push packages
       if: github.event_name == 'push' && github.ref == 'refs/heads/master'
-      run: dotnet nuget push **/*.nupkg -s https://f.feedz.io/davidfowl/featherhttp/nuget/index.json -ss https://f.feedz.io/davidfowl/featherhttp/symbols --skip-duplicate
\ No newline at end of file
+      run: dotnet nuget push **/*.nupkg -s https://f.feedz.io/davidfowl/featherhttp/nuget/index.json -ss https://f.feedz.io/davidfowl/featherhttp/symbols --skip-duplicate
+
+  generate-docs:
+    runs-on: ubuntu-latest    
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+      - name: generate gh-pages
+        if: github.event_name == 'push' && github.ref == 'refs/heads/master'
+        run: |
+          cd ./docs
+          npm install
+          npm run generate:gh-pages
+      - name: deploy
+        if: github.event_name == 'push' && github.ref == 'refs/heads/master'
+        uses: JamesIves/github-pages-deploy-action@releases/v3
```

Comment:
What is this?

Reply:
### the answer
This automates deployment of the static pages to a `gh-pages` branch. I limited the docs updates to have the same guards as the nuget push steps, so it won't build on a `pull_request`, only on a `push` to `master`

### justification 
I thought you might not want to publish manually from your local (although I do provide instructions on how to do it) so I put this ci job together. I've tested it from my fork and it appears to work pretty well.